apps/rush-lib/src/cli/actions/BaseRushAction.ts

```diff
@@ -113,4 +116,50 @@ export abstract class BaseRushAction extends BaseConfiglessRushAction {
     }
     return this._eventHooksManager;
   }
+
+  protected runRushCheckIfNecessary(isRushCheckCommand: boolean = false): void {
```

Comment:
>runRushCheckIfNecessary [](start = 12, length = 23)

This logic doesn't belong in BaseRushAction.  Maybe the helper can be here, but the algorithm itself should be moved to a more dedicated place.